src/hooks/useEscrowContractUsers.ts

```diff
@@ -0,0 +1,45 @@
+import { createOwner, TitleEscrowOwner } from "@govtechsg/oa-token";
+
+import { getLogger } from "../utils/logger";
+
+import { useEffect, useState } from "react";
+import { useEthereumTransactionState, TransactionStateStatus, StateType } from "./useEthereumTransactionState";
+import { useWeb3Provider } from "./useWeb3Provider";
+const { trace, error } = getLogger("hooks:useEscrowContractUsers");
+
+type UseEscrowContractUsers = { state: StateType; holderAddress: string; beneficiaryAddress: string };
+
+export const useEscrowContractUsers = ({
+  escrowContractAddress
+}: {
+  escrowContractAddress: string;
+}): UseEscrowContractUsers => {
+  trace(`escrow contract address ${escrowContractAddress}`);
+  const [holderAddress, setHolderAddress] = useState("");
+  const [beneficiaryAddress, setBeneficiaryAddress] = useState("");
+  const { web3Provider } = useWeb3Provider();
+  const [state, dispatch] = useEthereumTransactionState();
+
+  useEffect(() => {
+    async function fetchEscrowContractUsers() {
+      try {
+        dispatch({ type: TransactionStateStatus.LOADING });
+        const titleEscrowInstance = await createOwner({ address: escrowContractAddress, web3Provider });
+        if (!(titleEscrowInstance instanceof TitleEscrowOwner)) throw new Error("Address is not an Escrow contract");
```

Comment:
there is no other way to check this ?

Reply:
createOwner returns the class instance only. will keep it in mind if some how can be improved.

Reply:
there's a .isTitleEscrow() method on all Owner types, can use that
and also instead of throwing and catching an error can you just expose isTitleEscrow state as part of the hook? let the consumer of the hook decide what to do

Reply:
I dont think isTitleEscrow will help. As I am checking the class instance to verify if the instance is of the same class on which I am going to access member methods. to check for if address is title escrwo I will be creating a different hook useTokenOwnerInfo() which will be exposing the same state.

Reply:
https://github.com/Open-Attestation/oa-token/blob/748fd2a0b7201963e8604c6b1fffc0fde2b3d8d3/src/owner.ts#L63

this does exactly what you need